include/dlaf/auxiliary/norm/mc/norm_max_L.h

```diff
@@ -0,0 +1,107 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <hpx/hpx.hpp>
+#include <hpx/util/unwrap.hpp>
+
+#include "dlaf/common/vector.h"
+#include "dlaf/communication/communicator_grid.h"
+#include "dlaf/communication/sync/reduce.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/distribution.h"
+#include "dlaf/types.h"
+
+namespace dlaf {
+namespace internal {
+namespace mc {
+
+// Compute max norm of the lower triangular part of the distributed matrix
+// https://en.wikipedia.org/wiki/Matrix_norm#Max_norm
+//
+// It just addresses tiles with elements belonging to the lower triangular part of the matrix
+//
+// Thanks to the nature of the max norm, it is valid for:
+// - ge/sy/he lower
+// - tr lower non-unit
+template <class T>
+dlaf::BaseType<T> norm_max_L(comm::CommunicatorGrid comm_grid, Matrix<const T, Device::CPU>& matrix) {
+  using dlaf::common::internal::vector;
+  using dlaf::common::make_data;
+  using hpx::util::unwrapping;
+
+  using dlaf::tile::lange;
+  using dlaf::tile::lantr;
+
+  using NormT = dlaf::BaseType<T>;
+
+  const auto& distribution = matrix.distribution();
+
+  DLAF_ASSERT_SIZE_SQUARE(matrix);
+  DLAF_ASSERT_BLOCKSIZE_SQUARE(matrix);
+
+  if (GlobalElementSize{0, 0} == matrix.size())
+    return {0};
+
+  vector<hpx::future<NormT>> tiles_max;
+  tiles_max.reserve(distribution.localNrTiles().rows() * distribution.localNrTiles().cols());
+
+  // for each tile in local (lower triangular), create a task that finds the max element in the tile
```

Comment:
This comment may be misleading.
Maybe: `for each local tile in the (global) lower triangular matrix`